compiler/test/suites/functions.re

```diff
@@ -385,4 +385,173 @@ truc()|},
     |},
     "which has a default value, but the matching argument does not.",
   );
+
+  assertRun(
+    "partial1",
+    {|
+      let f = (x, y, z) => { print(x); print(y); print(z) }
+      let pf = partial f(1, _, _)
+      pf(2, 3)
+    |},
+    "1\n2\n3\n",
+  );
+  assertRun(
+    "partial2",
+    {|
+      let f = (x, y, z) => { print(x); print(y); print(z) }
+      let pf = partial f(_, 2, _)
+      pf(1, 3)
+    |},
+    "1\n2\n3\n",
+  );
+  assertRun(
+    "partial3",
+    {|
+      let f = (x, y, z) => { print(x); print(y); print(z) }
+      let pf = partial f(_, x=1, _)
+      pf(2, 3)
+    |},
+    "1\n2\n3\n",
+  );
+  assertRun(
+    "partial4",
+    {|
+      let f = (x, y, z) => { print(x); print(y); print(z) }
+      let pf = partial f(_, x=1, _)
+      pf(2, 3)
+    |},
+    "1\n2\n3\n",
+  );
+  assertRun(
+    "partial5",
+    {|
+      let f = (x, y, z) => { print(x); print(y); print(z) }
+      let pf = partial f(_, x=1, _)
+      pf(z=3, y=2)
+    |},
+    "1\n2\n3\n",
+  );
+  assertRun(
+    "partial6",
+    {|
+      let f = (x, y, z) => { print(x); print(y); print(z) }
+      let pf = partial f(1, _, y=_)
+      pf(3, 2)
+    |},
+    "1\n2\n3\n",
+  );
+  assertRun(
+    "partial6",
+    {|
+      let f = (x, y=2, z) => { print(x); print(y); print(z) }
+      let pf = partial f(1, _)
+      pf(3)
+    |},
+    "1\n2\n3\n",
+  );
+  assertRun(
+    "partial7",
+    {|
+      let f = (x, y=2, z) => { print(x); print(y); print(z) }
+      let pf = partial f(1, y=_, _)
+      pf(3)
+    |},
+    "1\n2\n3\n",
+  );
+  assertRun(
+    "partial8",
+    {|
+      let f = (x, y=0, z) => { print(x); print(y); print(z) }
+      let pf = partial f(1, y=_, _)
+      pf(y=2, 3)
+    |},
+    "1\n2\n3\n",
+  );
+  assertRun(
+    "partial9",
+    {|
+      let f = (x, y, _) => { print(x); print(y); print(3) }
+      let pf = partial f(1, y=_, _)
+      pf(y=2, 10)
+    |},
+    "1\n2\n3\n",
+  );
+  assertRun(
+    "partial10",
+    {|
+      let f = (x, y, z) => { print(x); print(y); print(z) }
+      let pf = (g, x) => partial g(x, _, _)
+      pf(f, 1)(2, 3)
+    |},
+    "1\n2\n3\n",
+  );
+  assertRun(
+    "partial11",
+    {|
+      let f = (x, y, z) => { print(x); print(y); print(z) }
+      let pf = (g, x) => partial g(1, b=x, _)
+      pf(f, 2)(3)
+    |},
+    "1\n2\n3\n",
+  );
+  assertRun(
+    "partial12",
+    {|
+      let f = (x, y=2, z) => { print(x); print(y); print(z) }
+      let makeFn = (x=0) => { print(x); f }
+      let pf = partial makeFn()(1, y=_, _)
+      pf(3)
+    |},
+    "0\n1\n2\n3\n",
+  );
+  assertRun(
+    "partial13",
+    {|
+      let (|>) = (x, f) => f(x)
```

Comment:
Should we add this function to pervasives? I feel like the moment we add partial application we are going to be asked for a pipeline operator soon after we add this.